sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/BuiltInFunctionResource.java

```diff
@@ -0,0 +1,173 @@
+public class BuiltInFunctionResource {
+
+  public static final String BASE_PATH = "/v2/namespaces/{namespace:\\w+}/collections";
+
+  @Inject FunctionService functionService;
+
+  @Inject TableManager tableManager;
+
+  @Operation(
+      description =
+          "Execute a built-in function (e.g. `$push` and `$pop`) against a value in this document. Performance may vary.")
+  @Parameters(
+      value = {
+        @Parameter(name = "namespace", ref = OpenApiConstants.Parameters.NAMESPACE),
+        @Parameter(name = "collection", ref = OpenApiConstants.Parameters.COLLECTION),
+        @Parameter(name = "document-id", ref = OpenApiConstants.Parameters.DOCUMENT_ID),
+        @Parameter(name = "document-path", ref = OpenApiConstants.Parameters.DOCUMENT_PATH),
+        @Parameter(name = "profile", ref = OpenApiConstants.Parameters.PROFILE),
+        @Parameter(name = "raw", ref = OpenApiConstants.Parameters.RAW),
+      })
+  @RequestBody(ref = OpenApiConstants.RequestBodies.FUNCTION)
+  @APIResponses(
+      value = {
+        @APIResponse(
+            responseCode = "200",
+            description =
+                "Function is executed. Certain functions will return a value in the `data` property. Note that in case of unwrapping (`raw=true`), the response contains only the contents of the `data` property.",
+            content = {
+              @Content(
+                  schema =
+                      @org.eclipse.microprofile.openapi.annotations.media.Schema(
+                          implementation = DocumentResponseWrapper.class),
+                  examples = @ExampleObject(ref = OpenApiConstants.Examples.FUNCTION_RESPONSE))
+            }),
+        @APIResponse(
+            responseCode = "404",
+            description = "Not found.",
+            content =
+                @Content(
+                    examples = {
+                      @ExampleObject(ref = OpenApiConstants.Examples.NAMESPACE_DOES_NOT_EXIST),
+                      @ExampleObject(ref = OpenApiConstants.Examples.COLLECTION_DOES_NOT_EXIST),
+                    },
+                    schema =
+                        @org.eclipse.microprofile.openapi.annotations.media.Schema(
+                            implementation = ApiError.class))),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_400),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_401),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_500),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_503),
+      })
+  @POST
+  @Path("{collection:\\w+}/{document-id}/{document-path:.*}/function")
+  public Uni<RestResponse<Object>> executeBuiltInFunction(
+      @Context UriInfo uriInfo,
+      @PathParam("namespace") String namespace,
+      @PathParam("collection") String collection,
+      @PathParam("document-id") String documentId,
+      @PathParam("document-path") List<PathSegment> documentPath,
+      @QueryParam("profile") boolean profile,
```

Comment:
Apologies for a newbie question but is this to get Profile information about execution (timing, operations etc)?
(I am guessing it is, based on `ExecutionContext` being created etc)
That's great if so, just wanted to verify.